Add paramter to loadstatus API to compute underdeplication against cluster view (#11056)
apache/druid: docs/operations/api-reference.md

Returns the number of segments left to load in each tier until segments that should be loaded in the cluster are all available. This includes segment replication counts.

* `/druid/coordinator/v1/loadstatus?full?computeUsingClusterView`

> Returns the number of segments left to load in each tier until segments that should be loaded in the cluster are all available.

Reviewer: ```suggestion
Returns the number of segments not yet loaded for each tier until all segments to load in the cluster are available.
```

> This includes segment replication counts. When computing the number of segments left to load, the number of nodes

Reviewer: ```suggestion
The result includes segment replication counts. It also factors in the number of available nodes that are of a service type that can load the segment when computing the number of segments remaining to load.
```

> available in the cluster that are of a service type that can load the segment is considered. A segment is considered fully

Reviewer: ```suggestion
A segment is considered fully loaded when:
```

> loaded if it is replicated as many times as configured in the corresponding load rule, or if there are as many replicas of

Reviewer: ```suggestion
- Druid had replicated it the number of times configured in the corresponding load rule.
- Or the number of replicas for the segment in each tier where it configured to be replicated equals the available nodes of a service type that are currently allowed to load the segment in the tier.
```

Reviewer 2: Suggest "equals the number of available nodes" instead in the suggested revision

> the segment in each tier where the segment is configured to be replicated, as there are available nodes in the tier at the

Reviewer: ```suggestion

```

> current time that are of a service type that is allowed to load the segment.

Reviewer: ```suggestion
```

Returns the number of segments left to load in each tier until segments that should be loaded in the cluster are all available. This includes segment replication counts.

* `/druid/coordinator/v1/loadstatus?full?computeUsingClusterView`

> Returns the number of segments not yet loaded for each tier until all segments to load in the cluster are available.

Reviewer: ```suggestion
Returns the number of segments not yet loaded for each tier until all segments loading in the cluster are available.
```

The result includes segment replication counts. It also factors in the number of available nodes that are of a service type that can load the segment when computing the number of segments remaining to load.
A segment is considered fully loaded when:

> - Druid had replicated it the number of times configured in the corresponding load rule.

Reviewer: ```suggestion
- Druid has replicated it the number of times configured in the corresponding load rule.
```

> - Or the number of replicas for the segment in each tier where it configured to be replicated equals the available nodes of a service type that are currently allowed to load the segment in the tier.

Reviewer: ```suggestion
- Or the number of replicas for the segment in each tier where it is configured to be replicated equals the available nodes of a service type that are currently allowed to load the segment in the tier.
```

(Note: `forceMetadataRefresh=true` refreshes Coordinator's metadata cache of all datasources. This can be a heavy operation in terms 
of the load on the metadata store but can be necessary to make sure that we verify all the latest segments' load status)
Setting `forceMetadataRefresh` to false will use the metadata cached on the coordinator from the last force/periodic refresh. 

> An optional query parameter `computeUsingClusterView` can be set to consider the cluster services available when computing

Reviewer: ```suggestion
You can pass the optional query parameter `computeUsingClusterView` to factor in the available cluster services when calculating
```

> the segments left to load. Please see [Coordinator Segment Loading](#coordinator-segment-loading) for a more detailed description of this parameter.

Reviewer: ```suggestion
the segments left to load. See [Coordinator Segment Loading](#coordinator-segment-loading) for details.
```
